File: 백준 문제모음/ATM.c
#include <stdio.h>
int main(void)
{
	int persons,i,j,temp,sum=0; //persons <- Danymic variable
	scanf("%d",&persons);
	int times[persons],res[persons];
	for(i=0;i<persons;i++)
	{
		scanf("%d",&times[i]);
	}
	//sort array
	for(i=0;i<persons-1;i++)
	{
		for(j=i+1;j<persons;j++)
		{
			if(times[i] > times[j])
			{
				temp = times[j];
				times[j] = times[i];
				times[i] = temp;
			}
		}
	}
	//sorted array times[persons]
	//each person total time
	//res[i] = times[0] + times[1] + ...times[i]
	for(i=0;i<persons;i++)
	{
		sum += times[i];
		res[i] = sum;
		//printf("No.%d person watinig about %dminutes.\n", i+1,res[i]);
	} 
	//sum initialize
	sum=0;
	//all people total times
	for(i=0;i<persons;i++)
	{
		sum+=res[i];
		//printf("%d\n", sum);
	} 
	printf("%d",sum);
	return 0;
}

File: 백준 문제모음/ATM.py
persons=int(input())
times=input()
list_each_time=times.rstrip().split(" ")
sum = 0
res = 0
sum_list=[]
if persons == len(list_each_time):
    if (1 <= persons <= 1000) :
        for i in range(len(list_each_time)):
            list_each_time[i] = int(list_each_time[i])
        list_each_time.sort()
        for i in range(len(list_each_time)):
            sum = sum + list_each_time[i]
            sum_list.append(sum)
        for itr in sum_list:
            res += itr
        print(res)
    else:
        print("Error!!")
else:
    print("Error!")



#출력 : 각 사람이 돈을 인출하는데 필요한 시간의 합의 최솟값을 출력한다.
 